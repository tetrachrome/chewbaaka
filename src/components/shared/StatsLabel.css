/**
 * StatsLabel.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Aug 12, 2020
 * Updated  : Aug 12, 2020
 */

div.ui.inverted.segment.StatsLabel {
  padding: 50px;
  text-align: center;
  border-radius: 10px;
  background-color: rgb(27,28,29);
}
