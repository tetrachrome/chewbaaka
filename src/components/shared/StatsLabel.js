/**
 * StatsLabel.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 25, 2020
 * Updated  : Jul 25, 2020
 */

import React from 'react'

import { Statistic, Segment } from 'semantic-ui-react'

import "semantic-ui-css/semantic.min.css"

export default function StatsLabel(props) {
  return (
    <Segment inverted>
      <Statistic inverted size="huge" color={props.color}>
        <Statistic.Value>{props.value}</Statistic.Value>
        <Statistic.Label>
          {props.label}
        </Statistic.Label>
      </Statistic>
    </Segment>
  );
}
