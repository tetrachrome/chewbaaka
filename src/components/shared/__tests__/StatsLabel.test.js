/**
 * StatsLabel.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 25, 2020
 * Updated  : Jul 25, 2020
 */

import React from 'react'

import { render, screen } from '@testing-library/react'

import renderer from 'react-test-renderer';

import StatsLabel from '../StatsLabel';

const value = "2020";
const label = "CCF turned 30";

test('renders StatsLabel component', () => {
  render(
    <StatsLabel
      value={value}
      label={label}
      color="orange"
    />
  );

  const valueElement = screen.getByText(value);
  expect(valueElement).toBeInTheDocument();

  const labelElement = screen.getByText(label);
  expect(labelElement).toBeInTheDocument();
});

test('StatsLabel component snapshot', () => {
  const tree = renderer
    .create(
      <StatsLabel
        value={value}
        label={label}
        color="orange"
      />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
