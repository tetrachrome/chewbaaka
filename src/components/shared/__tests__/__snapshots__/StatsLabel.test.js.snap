// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsLabel component snapshot 1`] = `
<div
  className="ui inverted segment"
>
  <div
    className="ui orange huge inverted statistic"
  >
    <div
      className="value"
    >
      2020
    </div>
    <div
      className="label"
    >
      CCF turned 30
    </div>
  </div>
</div>
`;
